run.go

```diff
@@ -53,51 +61,56 @@ func init() {
 	cmdRun.Flag.Var(&excludedPaths, "e", "Excluded paths[].")
 	cmdRun.Flag.BoolVar(&vendorWatch, "vendor", false, "Watch vendor folder")
 	cmdRun.Flag.StringVar(&buildTags, "tags", "", "Build tags (https://golang.org/pkg/go/build/)")
+	exit = make(chan bool)
 }
 
-var appname string
-
 func runApp(cmd *Command, args []string) int {
 	ShowShortVersionBanner()
 
-	exit := make(chan bool)
-	crupath, _ := os.Getwd()
-
 	if len(args) == 0 || args[0] == "watchall" {
-		appname = path.Base(crupath)
+		currpath, _ = os.Getwd()
+
+		if !isBeegoProject(currpath) {
+			exitPrint(fmt.Sprintf("Bee does not support non Beego project: %s", currpath))
+		}
```

Comment:
inside SearchGOPATHs already has isBeegoProject. I think you can judge by the `currentGoPath`, if it's empty it should be not Beego project